redshift_connector/core.py

```diff
@@ -658,7 +658,10 @@ def get_calling_module() -> str:
                 self._usock.setsockopt(socket.SOL_SOCKET, socket.SO_KEEPALIVE, 1)
         except socket.error as e:
             self._usock.close()
-            raise InterfaceError("communication error", e)
+            if socket.timeout:
```

Comment:
this check needs to be reworked as `socket.timeout` will always be true.  maybe something like this?

```python
except socket.error.timeout as timeout_e:
  raise OperationalError("connection time out", e)
except socket.error as e:
  raise InterfaceError("communication error", e)
```

Reply:
Hmm, I don't believe that's true. In manual testing, this error only comes up when I either disconnect my network, or when I set timeout = 0.

Reply:
You're right.... the evaluation of `if socket.timeout` will always return true. I'm now using two except blocks instead as you suggested. Thank you!